NitroxServer/GameLogic/Entities/EntityManager.cs

```diff
@@ -28,14 +28,46 @@ public class EntityManager
 
         public EntityManager(List<Entity> entities, BatchEntitySpawner batchEntitySpawner)
         {
-            entitiesById = entities.ToDictionary(entity => entity.Id);
+            entitiesById = new Dictionary<NitroxId, Entity>(entities.Count);
+            globalRootEntitiesById = new Dictionary<NitroxId, Entity>();
+            phasingEntitiesByAbsoluteCell = new Dictionary<AbsoluteEntityCell, List<Entity>>();
+            bool toWarn = false;
 
-            globalRootEntitiesById = entities.FindAll(entity => entity.ExistsInGlobalRoot)
-                                             .ToDictionary(entity => entity.Id);
-
-            phasingEntitiesByAbsoluteCell = entities.FindAll(entity => !entity.ExistsInGlobalRoot)
-                                                    .GroupBy(entity => entity.AbsoluteEntityCell)
-                                                    .ToDictionary(group => group.Key, group => group.ToList());
+            for (int i = 0; i < entities.Count; i++)
+            {
+                Entity entity = entities[i];
```

Comment:
Should we put this in the try catch incase something goes wrong with the index logic here and we out of index error?

Reply:
That's a valid option and would probably have an identical result.
Placing it directly next to the actual for loop is a strange habit I picked up because doing so with basic arrays actually causes a compiler optimization for accesses, although the impact of it is minimal and I don't think it works with List<T>.

As one additional comment that also applies to the foreach question:
List<T>.Count is a property that is always maintained, and the maximum index will always be 1 less than the Count property without fail, and the for loop will never iterate if it exceeds that number, even if it were to somehow be modified between loops.
The only circumstances such a thing would even happen is in cases where the equivalent foreach implementation would result in an immediate InvalidOperationException, due to an enumeration error.
Additionally, the only possible case for the Count to change between it being checked and the index being accessed would be if it were modified by multi-threaded code, which would cause far bigger problems since List<T> isn't thread safe in the first place.

Still, if it would be preferred to switch to a foreach, or to put the index access inside the try/catch, I'll change it.